src/model/RefrigerationSystem.cpp

```diff
@@ -174,6 +174,7 @@ namespace detail {
   {
     std::vector<IdfObject> result;
 
+    // TODO: technically this object can be shared accross multiple systems..
```

Comment:
Should be disallow sharing `RefrigerationCondenser` accross several RefrigerationSystem? Or not delete it unless it becomes orphaned? @kbenne ?

Reply:
@kbenne  an illustrative code example:

```ruby
m = Model.new
r = RefrigerationSystem.new(m)
c = RefrigerationCondenserAirCooled.new(m)
sub_liquid = RefrigerationSubcoolerLiquidSuction.new(m)
sub = RefrigerationSubcoolerMechanical.new(m)
r.setRefrigerationCondenser(c)
r.setMechanicalSubcooler(sub)
r.setLiquidSuctionHeatExchangerSubcooler(sub_liquid)

r2 = RefrigerationSystem.new(m)
r2.setRefrigerationCondenser(c)
r2.setMechanicalSubcooler(sub)
r2.setLiquidSuctionHeatExchangerSubcooler(sub_liquid)
puts r
```
```
OS:Refrigeration:System,
  {5f9fab10-4145-40b8-9360-f0af0ccf3747}, !- Handle
  Refrigeration System 1,                 !- Name
  {25e031bf-bdda-42ff-a216-2ff0f96c782b}, !- Refrigerated CaseAndWalkInList Name
  {398b1be7-9edd-4b77-abf5-eb78870a0310}, !- Refrigeration TransferLoad List Name
  {fa314522-b2e9-48d8-b5d5-0bd4f46bd746}, !- Refrigeration Condenser Name
  {677b3507-b338-45b0-9160-ec200ec39f7d}, !- CompressorList Name
  21,                                     !- Minimum Condensing Temperature {C}
  R404a,                                  !- Refrigeration System Working Fluid Type
  ConstantSuctionTemperature,             !- Suction Temperature Control Type
  {5199722d-3e8c-41c0-908c-e20a6021e16b}, !- Mechanical Subcooler Name
  {c18cbca6-ba57-4ba6-9401-489223a6aeb5}, !- Liquid Suction Heat Exchanger Subcooler Name
  ,                                       !- Sum UA Suction Piping {W/K}
  ,                                       !- Suction Piping Zone Name
  ,                                       !- End-Use Subcategory
  ,                                       !- Intercooler Type
  ,                                       !- Shell-and-Coil Intercooler Effectiveness
  {28610ebc-74f7-4803-a541-4fd3574f91ff}; !- High-Stage CompressorList Name
```

Remove r2, and now r loses all three components:

```
r2.remove
puts r
```
```
OS:Refrigeration:System,
  {5f9fab10-4145-40b8-9360-f0af0ccf3747}, !- Handle
  Refrigeration System 1,                 !- Name
  {25e031bf-bdda-42ff-a216-2ff0f96c782b}, !- Refrigerated CaseAndWalkInList Name
  {398b1be7-9edd-4b77-abf5-eb78870a0310}, !- Refrigeration TransferLoad List Name
  ,                                       !- Refrigeration Condenser Name
  {677b3507-b338-45b0-9160-ec200ec39f7d}, !- CompressorList Name
  21,                                     !- Minimum Condensing Temperature {C}
  R404a,                                  !- Refrigeration System Working Fluid Type
  ConstantSuctionTemperature,             !- Suction Temperature Control Type
  ,                                       !- Mechanical Subcooler Name
  ,                                       !- Liquid Suction Heat Exchanger Subcooler Name
  ,                                       !- Sum UA Suction Piping {W/K}
  ,                                       !- Suction Piping Zone Name
  ,                                       !- End-Use Subcategory
  ,                                       !- Intercooler Type
  ,                                       !- Shell-and-Coil Intercooler Effectiveness
  {28610ebc-74f7-4803-a541-4fd3574f91ff}; !- High-Stage CompressorList Name
```